cityofaustin/vision-zero
# Create crash records

## Associated issues

* https://github.com/cityofaustin/atd-data-tech/issues/19985
* https://github.com/cityofaustin/atd-data-tech/issues/19986

<img width="563" alt="Screenshot 2025-01-03 at 11 04 43 AM" src="https://github.com/user-attachments/assets/8f15bc2d-8788-467e-a0e9-aaaa974a0012" />

Note these two related features which are out of scope:

* Ability to delete temporary crash records - https://github.com/cityofaustin/vision-zero/pull/1644
* Hide this page from read-only users - https://github.com/cityofaustin/atd-data-tech/issues/20302


## Testing

**URL to test:** Local

1. Use the **Create crash** sidebar link to navigate to this new page
2. Use the **Create crash record** button to open the form. Create a few crash records while also testing the required fields validation. The required fields are Case ID, Crash date, primary address, and all three fields on each unit card.
3. Try adding and removing additional units while you're using the form.
4. Verify that the table refreshes after you submit the form.
5. Visit the crash details page and verify that the number and types of units created are correct, and that a person record is created for each injury.
6. On the temp crashers table, test the search and filter functionality.
7. I'd like to hide the filters altogether by adding a new prop called `staticFilters` (or something like that). I am holding off on that because there are other branches out there with changes to the table component's props.

---
#### Ship list
